File: app/src/app/dto/offers-dto.ts
import OffersInterface from "../interface/offers-interface";

export default class OffersDTO implements OffersInterface {
    userId: number;
    oldPrice: number;
    sneakerName: string;
    brand: string;
    link: string;
    newPrice: number;
    userName: string;
    chatId: number;

    constructor(data: Partial<OffersInterface>) {
        this.userId = data.userId || 0;
        this.oldPrice = data.oldPrice || 0;
        this.sneakerName = data.sneakerName || '';
        this.brand = data.brand || '';
        this.link = data.link || '';
        this.newPrice = data.newPrice || 0;
        this.userName = data.userName || '';
        this.chatId = data.chatId || 0;
    }
    
}

File: api/src/main/java/com/zeven/ets_proyect/Ets_Proyect/dto/sneakers/OffersDTO.java
package com.zeven.ets_proyect.Ets_Proyect.dto.sneakers;


import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class OffersDTO {

    private Long userId;
    private Integer oldPrice;
    private String sneakerName;
    private String brand;
    private String image;
    private String link;
    private Integer newPrice;
    private String userName;
    private Long chatId;
}
